packages/nx-betterer/src/lib/setup.ts

```diff
@@ -1,31 +1,43 @@
-import { ProjectGraphProjectNode, readJsonFile } from '@nrwl/devkit';
-import { readCachedProjectGraph } from '@nrwl/workspace/src/core/project-graph';
+import { readJsonFile } from '@nrwl/devkit';
+import { readWorkspace } from '@nrwl/devkit/src/generators/project-configuration';
+import { FsTree } from '@nrwl/tao/src/shared/tree';
+import { existsSync } from 'fs';
+import { dirname, join } from 'path';
 import type { NxBettererContext, TsConfig } from './types';
 
+/**
+ * Copy from @nrwl/nx
+ * https://github.com/nrwl/nx/blob/master/packages/tao/index.ts#L82-L94
+ */
+function findWorkspaceRoot(dir: string): string {
+    if (dirname(dir) === dir) {
+        throw new Error(`The cwd isn't part of an Nx workspace`);
+    }
+    if (
+        existsSync(join(dir, 'angular.json')) ||
+        existsSync(join(dir, 'workspace.json'))
+    ) {
+        return dir;
+    }
+    return findWorkspaceRoot(dirname(dir));
+}
+
 /**
  * Reads the configurations files that will be used in multiple places throughout the tests
  */
 export function setupNxBetterer(): NxBettererContext {
-    const plainWorkspaceJson =
-        readJsonFile<{ projects: Record<string, string | any> }>(
-            'workspace.json'
-        );
-
-    const workspaceJson = readCachedProjectGraph();
```

Comment:
Main of this contribution would be to not use internal nx project graph but rather read project configurations directly from FS.

* it will be faster as graph generation not required
* it will not require to invoke `nx graph` before `beterrer` (this could be solved by invoking graph generation inside setup if really needed)

Reply:
Also, please update the documentation in this PR such that adding the graph generation before hand is not longer needed.

Note: if we do need to generate the graph in the future it cannot be done in setup as that is an async operation and the betterer setup cannot be async, so it would need to be either a) done the way it currently is, or b) created in the tests themselves (as they can be async)